### PR TITLE
Remove SatisfiableRESTResource interface in favor for Java 8 default …

### PR DESCRIPTION
**Pull request by triller-telekom**
…interface methods

Part of Java 8 cleanup: https://github.com/eclipse/smarthome/pull/2320#issuecomment-253907800

Signed-off-by: Stefan Triller <stefan.triller@telekom.de>

### CODE REVIEW

File: bundles/io/org.eclipse.smarthome.io.rest/src/main/java/org/eclipse/smarthome/io/rest/RESTResource.java [hunk 1]
```diff
@@ -11,7 +11,17 @@
...
+ * @author Stefan Triller - Added default implementation for isSatisfied
  */
 public interface RESTResource {
 
+    /**
+     * Method used to determine availability of a RESTResource
+     *
+     * @return true if this RESTResource is ready to process requests, false otherwise.
+     */
+    default public boolean isSatisfied() {
```
Comment (maggu2810) [inline bundles/io/org.eclipse.smarthome.io.rest/src/main/java/org/eclipse/smarthome/io/rest/RESTResource.java]:
```
It is permitted, but discouraged as a matter of style, to redundantly specify the public and/or abstract modifier for a method declared in an interface.
```
http://docs.oracle.com/javase/specs/jls/se7/html/jls-9.html#jls-9.4

**Comment by kaikreuzer** [comment]
Note to implementors: All code using `SatisfiableRESTResource` needs to be refactored.

**Comment by BenediktNiehues** [comment]
@triller-telekom seems that you missed to adapt/remove the component descriptor satisfiableresourcefilter.xml in o.e.s.io.rest

**Comment by triller-telekom** [comment]
Thx for pointing this out, I created a new pr to remove it: #3452

**Pull request merged**